src/PureCloud.Client/Apis/WorkforceManagementApi.cs

```diff
@@ -0,0 +1,104 @@
+using System.Net.Http.Json;
+using Microsoft.Extensions.Options;
+using PureCloud.Client.Contracts;
+using PureCloud.Client.Http;
+using PureCloud.Client.Json;
+using PureCloud.Client.Models;
+
+namespace PureCloud.Client.Apis;
+
+public class WorkforceManagementApi : IWorkforceManagementApi
+{
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly PureCloudJsonSerializerOptions _options;
+
+    public WorkforceManagementApi(
+        IHttpClientFactory httpClientFactory,
+        IOptions<PureCloudJsonSerializerOptions> options)
+    {
+        _httpClientFactory = httpClientFactory;
+        _options = options.Value;
+    }
+
+    /// <summary>
+    /// Get a list of UserScheduleAdherence records for the requested users
+    /// </summary>
+    /// <param name="userIds">User Id(s) for which to fetch current schedule adherence information. Min 1, Max of 100 userIds per request</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>Array of UserScheduleAdherence records</returns>
+    public async Task<UserScheduleAdherence[]> GetAdherenceAsync(IEnumerable<string> userIds, CancellationToken cancellationToken = default)
+    {
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var parameters = new NameValueCollection();
+        if (userIds != null)
+        {
+            foreach (var id in userIds)
+            {
+                parameters.Add("userId", UriHelper.ParameterToString(id));
+            }
+        }
+
+        var requestUri = UriHelper.GetUri("api/v2/workforcemanagement/adherence", parameters);
+        var response = await client.GetAsync(requestUri, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        var result = await response.Content.ReadFromJsonAsync<List<UserScheduleAdherence>>(_options.JsonSerializerOptions, cancellationToken);
+        return result?.ToArray() ?? Array.Empty<UserScheduleAdherence>();
+    }
+
+    /// <summary>
+    /// Create adherence explanations query
+    /// </summary>
+    /// <param name="body">The query request body</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>Query adherence explanations response</returns>
+    public async Task<QueryAdherenceExplanationsResponse> CreateAdherenceExplanationsQueryAsync(AgentQueryAdherenceExplanationsRequest body, CancellationToken cancellationToken = default)
+    {
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var response = await client.PostAsJsonAsync("api/v2/workforcemanagement/adherence/explanations/query", body, _options.JsonSerializerOptions, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<QueryAdherenceExplanationsResponse>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <summary>
```

Comment:
Actually instead of duplicating the method documentation in the implementation just use /// <inheritdoc /> and keep the documenation in the interface.